ui/admin/app/components/model-providers/ModelProviderContext.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+    ReactNode,
+    createContext,
+    useCallback,
+    useContext,
+    useState,
+} from "react";
+import useSWR, { mutate } from "swr";
+
+import { ModelProvider } from "~/lib/model/modelProviders";
+import { ModelProviderApiService } from "~/lib/service/api/modelProviderApiService";
+
+import { useAsync } from "~/hooks/useAsync";
+
+interface ModelProviderContextType {
+    modelProviders: ModelProvider[];
+    configured: boolean;
+    configureModelProvider: (
+        modelProviderId: string,
+        value: Record<string, string>
+    ) => void;
+    isUpdating: boolean;
+    error?: unknown;
+    lastUpdated?: Date;
+}
```

Comment:
so it doesn't really look like much of this is reused (or really even used at all) so I'd argue that this doesn't really need to be put into a Context. My preference would be to just call `useSWR` in each place you need access to the model providers data